src/libdredd/src/mutation_replace_expr.cc

```diff
@@ -538,18 +538,18 @@ void MutationReplaceExpr::ReplaceExprWithFunctionCall(
     suffix.append("; }");
   }
 
-  if (!ast_context.getLangOpts().CPlusPlus) {
-    if (expr_->isLValue() && input_type.ends_with('*')) {
-      prefix.append("&(");
-      suffix.append(")");
-    } else if (const auto* binary_expr =
-                   llvm::dyn_cast<clang::BinaryOperator>(expr_)) {
-      // The comma operator requires special care in C, to avoid it appearing to
-      // provide multiple parameters for an enclosing function call.
-      if (binary_expr->isCommaOp()) {
-        prefix.append("(");
-        suffix.append(")");
-      }
+  if (!ast_context.getLangOpts().CPlusPlus && expr_->isLValue() &&
+      input_type.ends_with('*')) {
+    prefix.append("&(");
+    suffix = ")" + suffix;
+  }
+
+  if (const auto* binary_expr = llvm::dyn_cast<clang::BinaryOperator>(expr_)) {
+    // The comma operator requires special care in C, to avoid it appearing to
+    // provide multiple parameters for an enclosing function call.
```

Comment:
Can you remove the comment about "in C"?

I think that initially, because we did everything using lambdas for C++, this was not a problem for C++. But now that we try to avoid using lambdas where possible, I see that it's also a problem for C++.